### PR TITLE
refactor(next): shared sidebar helpers - page-chrome parity to prevent stable/next drift (#1944)

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

Establishes a page-chrome parity pattern for dual-channel screens to prevent stable/next sidebar control drift, by extracting the sidebar display helpers into one shared package used by both channels.

- New `web/templates/sidebar_helpers.go` (package `templates`) with `SidebarInitial` and `SidebarDisplayName`, imported by both the stable (`web/templates/`) and next (`web/templates/next/`) sidebars.
- `SidebarInitial` now uppercases the first rune on **both** channels (adopting the next-channel behavior), returning `?` for empty/whitespace input - fixing the drift where the stable avatar initial rendered lowercase.
- Removes the duplicated next-only helper (`web/templates/next/sidebar_helpers.go`) and consolidates the test.

## Verification (lead-driven UAT)

Rendered on a live instance (both channels, same DB):
- Stable `/` sidebar avatar = **"A"** (uppercase), display name "admin".
- Next `/next/` sidebar avatar = **"A"** (uppercase), display name "admin".
- Match confirmed - the shared helper produces identical output on both channels; stable initials are now uppercase as intended. Screenshot captured.

Closes #1944

## Summary by CodeRabbit

* **Refactor**
  * Reorganized sidebar helper functions from next-specific package into a shared templates package to improve code reuse and maintainability across components.

* **Tests**
  * Added comprehensive unit tests for sidebar display name formatting and avatar initial character derivation, with coverage of edge cases and Unicode handling.

* **Documentation**
  * Enhanced template documentation with clarifying comments describing chrome composition patterns and sidebar helper function location references.

### CODE REVIEW

File: web/templates/sidebar_helpers.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+package templates
+
+import "strings"
+
+// SidebarInitial returns the uppercased first rune of s, or "?" when s is
+// empty or whitespace-only. Adopts the next-channel behavior (uppercase) for
+// visual consistency with conventional avatar initials.
+func SidebarInitial(s string) string {
+	for _, r := range strings.TrimSpace(s) {
+		return strings.ToUpper(string(r))
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply web/templates/sidebar_helpers.go]: @codoki Correct edge case. Switched to `string(unicode.ToUpper(r))` so the avatar initial stays a single code point (e.g. 'ß' no longer expands to "SS"); added the `unicode` import (kept `strings` for TrimSpace) and extended sidebar_helpers_test.go with the multi-rune-uppercase case plus a single-glyph guard (utf8.RuneCountInString==1) and the whitespace/empty fallbacks. Fixed in `3162c7f6`.
